socorro/external/es/crashstorage.py

```diff
@@ -46,6 +46,39 @@ def __init__(self, config):
 VALID_KEY = re.compile(r'^[a-zA-Z0-9_-]+$')
 
 
+# Cache of (fields id, analyzer) -> list of fields properties
+_ANALYZER_TO_FIELDS_MAP = {}
+
+
+def get_fields_by_analyzer(fields, analyzer):
+    """Returns the fields in fields that have the specified analyzer
+
+    Note: This "hashes" the fields argument by using `id`. I think this is fine
+    because fields doesn't change between runs and it's not mutated in-place.
+    We're we're hashing it sufficiently often that it's faster to use `id` than
+    a more computationally intensive hash of a large data structure.
+
+    :arg dict fields: dict of field information mapped as field name to
+        properties
+    :arg str analyzer: the Elasticsearch analyzer to match
+
+    :returns: list of field properties for fields that match the analyzer
+
+    """
+    map_key = (id(fields), analyzer)
+    try:
+        return _ANALYZER_TO_FIELDS_MAP[map_key]
+    except KeyError:
+        pass
+
+    fields = [
+        field for field in fields.values()
+        if (field.get('storage_mapping') or {}).get('analyzer', '') == analyzer
+    ]
+    _ANALYZER_TO_FIELDS_MAP[map_key] = fields
+    return fields
```

Comment:
I needed this kind of thing for `convert_booleans` and figured I'd extract it from `truncate_keyword_field_values`. That makes it easier to test in isolation, too.